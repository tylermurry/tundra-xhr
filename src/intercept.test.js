import interceptCalls from './intercept';
import submitRequestData from './submitRequest';

function makeFakeRequest() {
  // It's not possible to both mock XHR and make a 'fake' real request.
  // Instead, we're calling out to a real URL that will likely always be
  // there and unlikely to change. #tradeoffs

  return new Promise(((resolve) => {
    const xhr = new XMLHttpRequest(); // eslint-disable-line no-undef

    xhr.open('GET', 'http://httpstat.us/200');
    xhr.onload = resolve;
    xhr.send();
  }));
}

jest.mock('./submitRequest', () => jest.fn());

describe('intercept', () => {
  it('should proxy the request properly', async () => {
    const callback = jest.fn();

    interceptCalls(undefined, callback);

    await makeFakeRequest();

    // Dear future self, please forgive this mortal sin. Fix when able.
    callback.mock.calls[0][0].response.headers['date'] = 'static date';
    submitRequestData.mock.calls[0][0].response.headers['date'] = 'static date';

    expect(callback.mock.calls).toMatchSnapshot();
    expect(submitRequestData.mock.calls).toMatchSnapshot();
  });
});
