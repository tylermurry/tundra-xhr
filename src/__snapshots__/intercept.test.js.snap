// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`intercept should proxy the request properly 1`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": null,
        "headers": Object {},
        "method": "GET",
        "url": "http://httpstat.us/200",
      },
      "response": Object {
        "content": "",
        "headers": Object {
          "access-control-allow-origin": "*",
          "cache-control": "private",
          "connection": "close",
          "content-length": "0",
          "date": "static date",
          "server": "Microsoft-IIS/10.0",
          "set-cookie": "ARRAffinity=beec3692495883b8df6195e900c12f49514e054d865a22ad2951c84f51dbaf93;Path=/;HttpOnly;Domain=httpstat.us",
          "x-aspnet-version": "4.0.30319",
          "x-aspnetmvc-version": "5.1",
          "x-powered-by": "ASP.NET",
        },
        "statusCode": 200,
      },
    },
  ],
]
`;

exports[`intercept should proxy the request properly 2`] = `
Array [
  Array [
    Object {
      "request": Object {
        "content": null,
        "headers": Object {},
        "method": "GET",
        "url": "http://httpstat.us/200",
      },
      "response": Object {
        "content": "",
        "headers": Object {
          "access-control-allow-origin": "*",
          "cache-control": "private",
          "connection": "close",
          "content-length": "0",
          "date": "static date",
          "server": "Microsoft-IIS/10.0",
          "set-cookie": "ARRAffinity=beec3692495883b8df6195e900c12f49514e054d865a22ad2951c84f51dbaf93;Path=/;HttpOnly;Domain=httpstat.us",
          "x-aspnet-version": "4.0.30319",
          "x-aspnetmvc-version": "5.1",
          "x-powered-by": "ASP.NET",
        },
        "statusCode": 200,
      },
    },
    9091,
    true,
    undefined,
  ],
]
`;
